### PR TITLE
Add basic authentication jira v3

### PR DESCRIPTION
**Pull request by israelpoli**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-7285

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Jira/Integrations/JiraV3/JiraV3.py [hunk 1]
```diff
@@ -114,25 +117,41 @@ def test_instance_connection(self) -> None:
         their own connectivity test
         """
 
-    def http_request_with_access_token(self, method: str, headers: Dict[str, str] | None = None, url_suffix='', params=None,
-                                       data=None, json_data=None, resp_type='json', ok_codes=None, full_url='',
-                                       files: Dict[str, Any] | None = None) -> Any:
-        """This method wraps the _http_request that comes from the BaseClient class, and adds the access_token
-        of the client to the headers request, by calling the get_access_token method, which is an abstract method,
-        and every child class must implement it.
+    def http_request(self, method: str, headers: dict[str, str] = {}, url_suffix='', params=None,
```
Comment (anas-yousef) [inline Packs/Jira/Integrations/JiraV3/JiraV3.py:120]: Adding a default empty dictionary for an argument, **headers** in our case, can cause unwanted behavior, as explained [here](https://stackoverflow.com/questions/366422/how-can-i-avoid-issues-caused-by-pythons-early-bound-default-parameters-e-g-m), where the headers object can "remember" old data.

File: Packs/Jira/Integrations/JiraV3/JiraV3.py [hunk 2]
```diff
@@ -3810,18 +3838,44 @@ def get_issue_id_or_key(issue_id: str = '', issue_key: str = '') -> str:
     return issue_id or issue_key
 
 
+def validate_auth_params(
+    username: str, api_key: str, client_id: str, client_secret: str
+) -> None:
+    basic_oauth, oauth2 = username or api_key, client_id or client_secret
```
Comment (dorschw) [inline Packs/Jira/Integrations/JiraV3/JiraV3.py:3844]: the variable names`basic_outh` and `oauth2` are confusing, consider `credential_key`, `credential_value` or something similar

Comment (dorschw) [inline Packs/Jira/Integrations/JiraV3/JiraV3.py:3844]: let's discuss this line offline

File: Packs/Jira/Integrations/JiraV3/JiraV3.yml [hunk 3]
```diff
@@ -20,12 +25,12 @@ configuration:
...
   display: Callback URL
   type: 0
-  required: true
+  required: false
 - display: Client ID
   type: 9
   name: credentials
   displaypassword: Client Secret
-  required: true
+  required: false
```
[the remark this answers is not in the document]
Comment (israelpoli) [reply Packs/Jira/Integrations/JiraV3/JiraV3.yml:33]: Following the addition of the `basic auth` authentication method, the `Client ID` and `Client Secret` fields are not required,
Added validation that checks the parameters

File: Packs/Jira/Integrations/JiraV3/JiraV3.yml [hunk 4]
```diff
@@ -20,12 +25,12 @@ configuration:
 - name: callback_url
   display: Callback URL
   type: 0
-  required: true
+  required: false
```
[the remark this answers is not in the document]
Comment (israelpoli) [reply Packs/Jira/Integrations/JiraV3/JiraV3.yml:28]: Following the addition of the `basic auth` authentication method, the `Callback URL` field is not required,
Added validation that checks the parameters

File: Packs/Jira/Integrations/JiraV3/JiraV3.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (dorschw) [reply Packs/Jira/Integrations/JiraV3/JiraV3.yml:33]: nice

File: Packs/Jira/Integrations/JiraV3/JiraV3_description.md [hunk 5]
```diff
@@ -3,38 +3,60 @@ Please configure only one of the following fields:
 1. Cloud ID - Used for Jira Cloud instance.
 2. OnPrem - Leave the Cloud ID empty, and fill in the rest of the fields.
 
-For both instances, it is advised to use the `https://oproxy.demisto.ninja/authcode` **Callback URL**. The oproxy url is a client side only web page which provides an easy interface to copy the obtained auth code from the authorization response to the integration configuration in the authorization flow steps. Optionally: if you don't want to use the oproxy url, you may use a localhost url on a port which is not used locally on your machine. For example: <http://localhost:9004>. You will then need to copy the code from the url address bar in the response.
+##### Cloud ID
 
-### Cloud authentication
+Go to your [Admin page](https://admin.atlassian.com/), click the **Products** tab on the top banner and choose the appropriate site under **Sites and Products** on the left side bar. Your Cloud ID will appear in the URL:
```
Comment (ShirleyDenkberg) [inline Packs/Jira/Integrations/JiraV3/JiraV3_description.md:8]:
```suggestion
1. Go to your [Admin page](https://admin.atlassian.com/.
2. Click the **Products** tab on the top banner and choose the appropriate site under **Sites and Products** on the left side bar. Your Cloud ID will appear in the URL:
```

File: Packs/Jira/Integrations/JiraV3/JiraV3_description.md [hunk 6]
```diff
@@ -3,38 +3,60 @@ Please configure only one of the following fields:
...
+`https://admin.atlassian.com/s/{cloud_id}/users`
+
+#### Authentication Methods
+There are 2 authentication methods:
+- **Basic Authentication**
+- **OAuth 2.0**
+
+### Basic Authentication
+Leave the *Client ID* and *Client Secret* fields empty and fill in the following fields:
+- *User name* - enter the your user email
```
Comment (ShirleyDenkberg) [inline Packs/Jira/Integrations/JiraV3/JiraV3_description.md:18]:
```suggestion
- *User name* - Enter your user email.
```

File: Packs/Jira/Integrations/JiraV3/JiraV3_description.md [hunk 7]
```diff
@@ -3,38 +3,60 @@ Please configure only one of the following fields:
...
+- *API key* - enter the API token, to generate API token see [here](https://support.atlassian.com/atlassian-account/docs/manage-api-tokens-for-your-atlassian-account/)
```
Comment (ShirleyDenkberg) [inline Packs/Jira/Integrations/JiraV3/JiraV3_description.md:19]:
```suggestion
- *API key* - Enter the API token. To generate API token, see [here](https://support.atlassian.com/atlassian-account/docs/manage-api-tokens-for-your-atlassian-account/)
```

File: Packs/Jira/Integrations/JiraV3/JiraV3_description.md [hunk 8]
```diff
@@ -3,38 +3,60 @@ Please configure only one of the following fields:
...
+
+##### Basic Authentication permissions
+Ensure that you possess the necessary permissions by navigating to **Project Settings** -> **Permissions** in the project.
```
Comment (ShirleyDenkberg) [inline Packs/Jira/Integrations/JiraV3/JiraV3_description.md:22]:
```suggestion
Ensure that you possess the necessary permissions by navigating to **Project Settings** > **Permissions** in the project.
```

File: Packs/Jira/Integrations/JiraV3/README.md [hunk 9]
```diff
@@ -38,40 +40,65 @@ If you are upgrading from a previous version of this integration, see [Breaking
 
 5. Check [Authorization Flow In Cortex XSOAR](#authorization-flow-in-cortex-xsoar) in order to authenticate and test the connection.
 
+
+Please configure only one of the following fields:
+
+1. Cloud ID - Used for Jira Cloud instance.
+2. OnPrem - Leave the Cloud ID empty, and fill in the rest of the fields.
```
Comment (ShirleyDenkberg) [inline Packs/Jira/Integrations/JiraV3/README.md:47]:
```suggestion
- Cloud ID - Used for Jira Cloud instance.
- OnPrem - Leave the Cloud ID empty, and fill in the rest of the fields.
```

File: Packs/Jira/Integrations/JiraV3/README.md [hunk 10]
```diff
@@ -38,40 +40,65 @@ If you are upgrading from a previous version of this integration, see [Breaking
...
+
+##### Cloud ID
+
+Go to your [Admin page](https://admin.atlassian.com/), click the **Products** tab on the top banner and choose the appropriate site under **Sites and Products** on the left side bar. Your Cloud ID will appear in the URL:
```
Comment (ShirleyDenkberg) [inline Packs/Jira/Integrations/JiraV3/README.md:51]:
```suggestion
1. Go to your [Admin page](https://admin.atlassian.com/).
2. Click the **Products** tab on the top banner and choose the appropriate site under **Sites and Products** on the left side bar. Your Cloud ID will appear in the URL:
```

File: Packs/Jira/Integrations/JiraV3/README.md [hunk 11]
```diff
@@ -38,40 +40,65 @@ If you are upgrading from a previous version of this integration, see [Breaking
...
+`https://admin.atlassian.com/s/{cloud_id}/users`
+
 ## Authentication
+### Basic Authentication
+Leave the *Client ID* and *Client Secret* fields empty and fill in the following fields:
+- *User name* - enter the your user email
```
Comment (ShirleyDenkberg) [inline Packs/Jira/Integrations/JiraV3/README.md:57]:
```suggestion
- *User name* -Enter your user email.
```

File: Packs/Jira/Integrations/JiraV3/README.md [hunk 12]
```diff
@@ -38,40 +40,65 @@ If you are upgrading from a previous version of this integration, see [Breaking
...
+- *API key* - enter the API token, to generate API token see [here](https://support.atlassian.com/atlassian-account/docs/manage-api-tokens-for-your-atlassian-account/)
```
Comment (ShirleyDenkberg) [inline Packs/Jira/Integrations/JiraV3/README.md:58]:
```suggestion
- *API key* - Enter the API token. To generate API token, see [here](https://support.atlassian.com/atlassian-account/docs/manage-api-tokens-for-your-atlassian-account/)
```

File: Packs/Jira/Integrations/JiraV3/README.md [hunk 13]
```diff
@@ -38,40 +40,65 @@ If you are upgrading from a previous version of this integration, see [Breaking
...
 
-For both instances, it is advised to use the `https://oproxy.demisto.ninja/authcode` **Callback URL**. The oproxy url is a client side only web page which provides an easy interface to copy the obtained auth code from the authorization response to the integration configuration in the authorization flow steps. Optionally: if you don't want to use the oproxy url, you may use a localhost url on a port which is not used locally on your machine. For example: <http://localhost:9004>. You will then need to copy the code from the url address bar in the response (see [Authorization Flow In Cortex XSOAR](#authorization-flow-in-cortex-xsoar)).
+##### Basic Authentication permissions
+Ensure that you possess the necessary permissions by navigating to **Project Settings** -> **Permissions** in the project.
```
Comment (ShirleyDenkberg) [inline Packs/Jira/Integrations/JiraV3/README.md:61]:
```suggestion
Ensure that you possess the necessary permissions by navigating to **Project Settings** > **Permissions** in the project.
```

File: Packs/Jira/Integrations/JiraV3/README.md [hunk 14]
```diff
@@ -38,40 +40,65 @@ If you are upgrading from a previous version of this integration, see [Breaking
...
+* Transition issues
+* Create attachments
+* Add comments
+* Link issues
 
-### Cloud authentication
+
+
+### OAuth 2.0
+For both instances (Cloud ID & OnPrem), it is advised to use the `https://oproxy.demisto.ninja/authcode` **Callback URL**. The oproxy url is a client side only web page which provides an easy interface to copy the obtained auth code from the authorization response to the integration configuration in the authorization flow steps. Optionally: if you don't want to use the oproxy url, you may use a localhost url on a port which is not used locally on your machine. For example: <http://localhost:9004>. You will then need to copy the code from the url address bar in the response (see [Authorization Flow In Cortex XSOAR](#authorization-flow-in-cortex-xsoar)).
```
Comment (ShirleyDenkberg) [inline Packs/Jira/Integrations/JiraV3/README.md:75]:
```suggestion
For both instances (Cloud ID & OnPrem), it is advised to use the `https://oproxy.demisto.ninja/authcode` **Callback URL**. The OProxy URL is a client side only web page which provides an easy interface to copy the obtained auth code from the authorization response to the integration configuration in the authorization flow steps. Optionally, if you don't want to use the OProxy URL, you can use a localhost URL on a port that is not used locally on your machine. For example: <http://localhost:9004>. You will then need to copy the code from the URL address bar in the response (see [Authorization Flow In Cortex XSOAR](#authorization-flow-in-cortex-xsoar)).
```

File: Packs/Jira/ReleaseNotes/3_1_0.md [hunk 15]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Atlassian Jira v3
+
+- Added support for *Basic Authentication* and used for both Cloud and On-premise.
```
Comment (ShirleyDenkberg) [inline Packs/Jira/ReleaseNotes/3_1_0.md:6]:
```suggestion
Added support for *Basic Authentication* used for both Cloud and On-premise.
```

File: Packs/Jira/Integrations/JiraV3/JiraV3.py [hunk 16]
```diff
@@ -3810,18 +3835,44 @@ def get_issue_id_or_key(issue_id: str = '', issue_key: str = '') -> str:
...
+
+    if (not basic_oauth) and (not oauth2):
+        raise DemistoException("The required parameters were not provided, see the help window")
```
Comment (ShirleyDenkberg) [inline Packs/Jira/Integrations/JiraV3/JiraV3.py:3844]:
```suggestion
        raise DemistoException("The required parameters were not provided. See the help window for more information.")
```

File: Packs/Jira/Integrations/JiraV3/JiraV3.py [hunk 17]
```diff
@@ -3810,18 +3835,44 @@ def get_issue_id_or_key(issue_id: str = '', issue_key: str = '') -> str:
...
+    if basic_oauth and oauth2:
+        raise DemistoException("The `User name` or `API key` parameters cannot be provided together"
+                               " with the `Client ID` or `Client Secret` parameters, see the help window")
```
Comment (ShirleyDenkberg) [inline Packs/Jira/Integrations/JiraV3/JiraV3.py:3847]:
```suggestion
                               " with the `Client ID` or `Client Secret` parameters. See the help window for more information.")
```

File: Packs/Jira/Integrations/JiraV3/JiraV3.py [hunk 18]
```diff
@@ -3810,18 +3835,44 @@ def get_issue_id_or_key(issue_id: str = '', issue_key: str = '') -> str:
...
+    if basic_oauth and not (username and api_key):
+        raise DemistoException(
+            "To use basic authentication, the 'User name' and 'API key' parameters are mandatory"
```
Comment (ShirleyDenkberg) [inline Packs/Jira/Integrations/JiraV3/JiraV3.py:3850]:
```suggestion
            "To use basic authentication, the 'User name' and 'API key' parameters are mandatory."
```

File: Packs/Jira/Integrations/JiraV3/JiraV3.py [hunk 19]
```diff
@@ -3810,18 +3835,44 @@ def get_issue_id_or_key(issue_id: str = '', issue_key: str = '') -> str:
...
+        )
+    if oauth2 and not (client_id and client_secret):
+        raise DemistoException(
+            "To use OAuth 2.0, the 'Client ID' and 'Client Secret' parameters are mandatory"
```
Comment (ShirleyDenkberg) [inline Packs/Jira/Integrations/JiraV3/JiraV3.py:3854]:
```suggestion
            "To use OAuth 2.0, the 'Client ID' and 'Client Secret' parameters are mandatory."
```

File: Packs/Jira/Integrations/JiraV3/JiraV3.py [hunk 20]
```diff
@@ -2954,16 +2973,22 @@ def test_authorization(client: JiraBaseClient, args: Dict[str, Any]) -> CommandR
...
-                           ' `code`, and paste that value in the command `!jira-ouath-complete` as an argument to finish'
-                           ' the process.')
+    if client.is_basic_auth:
+        client.test_instance_connection()  # raises on failure
+        return "ok"
+    else:
+        raise DemistoException(
+            'In order to authorize the instance, first run the command `!jira-oauth-start`,'
+            ' and complete the process in the URL that is returned. You will then be redirected'
+            ' to the callback URL . Copy the authorization code found in the query parameter'
```
Comment (ShirleyDenkberg) [inline Packs/Jira/Integrations/JiraV3/JiraV3.py:2988]:
```suggestion
            ' to the callback URL. Copy the authorization code found in the query parameter'
```

File: Packs/Jira/Integrations/JiraV3/JiraV3_test.py [hunk 21]
```diff
@@ -228,6 +228,157 @@ def test_get_issue_id_or_key(issue_id, issue_key, expected_issue_id_or_key):
...
+        - run `test_module` function
+    Then:
+        - Ensure that error msg is raised, with a guide how to connect through oauth2
+    """
+    from JiraV3 import test_module
+    client = jira_base_client_mock()
+    mocker.patch.object(client, "test_instance_connection")
+    with pytest.raises(
+        DemistoException,
+        match="In order to authorize the instance, first run the command `!jira-oauth-start`"
```
Comment (ShirleyDenkberg) [inline Packs/Jira/Integrations/JiraV3/JiraV3_test.py:295]:
```suggestion
        match="In order to authorize the instance, first run the command `!jira-oauth-start`."
```

**Pull request merged**